src/pages/student_fees_list/FeesList.tsx

```diff
@@ -0,0 +1,277 @@
+import { SortableButton } from "../../components/sort_button/SortButton";
+import {
+  StudentsContainer,
+  FiltersContainer,
+  LeftContainer,
+  RightContainer,
+  ActionButton,
+  ViewIconStyle,
+  FeeStatus,
+  FeeStatusContainer,
+  InformationStudent,
+  Title,
+  TitleContainer,
+  CoinIconStyles,
+  SubTitle,
+} from "./FeesList.styles";
+import Filter from "../../components/filter/Filter";
+import Button from "../../components/button/Button";
+import Table from "../../components/table/Table";
+import type { Column } from "../../app/types/table";
+import AddIcon from "../../assets/add-icon.svg";
+import BackIcon from "../../assets/back-icon.svg";
+import StudentIcon from "../../assets/student-icon.svg";
+import ViewIcon from "../../assets/openEye-icon.png";
+import CoinIcon from "../../assets/coin-icon.svg";
+import type { StudentMonthlyFeeResponse } from "../../app/types/responses/StudentMonthlyFee.type";
+import { useState } from "react";
+import { ConfirmDialog } from "../../components/confirm_dialog/ConfirmDialog";
+import {
+  useGetStudentByIdQuery,
+  useGetStudentMonthlyFeesQuery,
+} from "../../app/services/StudentService";
+import { skipToken } from "@reduxjs/toolkit/query/react";
+import { translateMonth } from "../../utils/TranslateMonths";
+import { formatCurrency } from "../../utils/Formatter";
+import DateFilter from "../../components/date_filter/DateFilter";
+import { toast } from "react-hot-toast";
+import { useUpdateMonthlyFeeMutation } from "../../app/services/MonthlyFeeService";
+import PaymentInfoModal from "../../components/payment_detail/paymentInfo";
+
+function StudentFeesList() {
+  const location = useLocation();
+  const { student } = location.state || {};
+  const { data: studentDetails } = useGetStudentByIdQuery(student.id);
+  const navigate = useNavigate();
+  const [monthFilter, setMonthFilter] = useState("");
+  const [expirationDateFilter, setExpirationDateFilter] = useState<Date | null>(
+    null
+  );
+  const [statusFilter, setStatusFilter] = useState("");
+  const [modalType, setModalType] = useState<"pay" | "details" | null>(null);
+  const [selectedFeeId, setSelectedFeeId] = useState<string | null>(null);
+  const [selectedPaymentId, setSelectedPaymentId] = useState<string | null>(
+    null
+  );
+  const [payMonthlyFee] = useUpdateMonthlyFeeMutation();
+
+  const normalizeDate = (date: Date | string | null) => {
+    if (!date) return undefined;
+
+    const d = new Date(date);
+
+    if (isNaN(d.getTime())) return undefined;
+    return d.toISOString().split("T")[0]; // "YYYY-MM-DD"
+  };
+  const formattedExpirationDate = normalizeDate(expirationDateFilter);
+  const {
+    data: fees,
+    isLoading,
+    isError,
+  } = useGetStudentMonthlyFeesQuery(
+    student?.id
+      ? {
+          studentId: student.id,
+          month: monthFilter || undefined,
+          expirationDate: formattedExpirationDate,
+          status: statusFilter || undefined,
+        }
+      : skipToken
+  );
```

Comment:
Y acá dos cosas:

1. Para qué seteamos el `formattedExpirationDate` si solo lo utilizamos en un unico lugar?
2. Podemos hacer lo siguiente para pasar la fecha de expiracion


```suggestion
  const {
    data: fees,
    isLoading,
    isError,
  } = useGetStudentMonthlyFeesQuery(
    student?.id
      ? {
          studentId: student.id,
          month: monthFilter || undefined,
          expirationDate: expirationDateFilter && normalizeDate(expirationDateFilter),
          status: statusFilter || undefined,
        }
      : skipToken
  );
```